zerocopy-derive/src/lib.rs

```diff
@@ -195,16 +195,20 @@ fn derive_from_bytes_union(ast: &DeriveInput, unn: &DataUnion) -> proc_macro2::T
 // - `repr(packed)`
 
 fn derive_as_bytes_struct(ast: &DeriveInput, strct: &DataStruct) -> proc_macro2::TokenStream {
-    // TODO(#10): Support type parameters.
-    if !ast.generics.params.is_empty() {
+    let reprs = try_or_print!(STRUCT_UNION_AS_BYTES_CFG.validate_reprs(ast));
+    let is_transparent = reprs.contains(&StructRepr::Transparent);
+
+    // TODO(#10): Support type parameters for non-transparent structs.
+    if !ast.generics.params.is_empty() && !is_transparent {
         return Error::new(Span::call_site(), "unsupported on types with type parameters")
             .to_compile_error();
     }
 
-    let reprs = try_or_print!(STRUCT_UNION_AS_BYTES_CFG.validate_reprs(ast));
-    let padding_check =
-        if reprs.contains(&StructRepr::Packed) { PaddingCheck::None } else { PaddingCheck::Struct };
-
+    let padding_check = if is_transparent || reprs.contains(&StructRepr::Packed) {
```

Comment:
Can you add a comment explaining why `is_transparent` implies that we don't need a padding check? Off the top of my head, this hangs on two factors (but maybe there's more I'm missing that needs to go into a justification):
- Rust itself will enforce that `repr(transparent)` only compiles if there's exactly one non-ZST field
- `repr(transparent)` means that the layout is identical to that of the non-ZST type, so `AsBytes` validity here is equivalent to `AsBytes` validity for that type, which is representable with a trait bound

Reply:
Yep, and both of those are true. I left a comment that I think conveys that.